image/image.go

```diff
@@ -325,6 +356,11 @@ func bootstrapToRootDir(tsto *ToolingStore, model *asserts.Model, opts *Options,
 		}
 	}
 
+	// ensure kernel-track is honored
+	if err := amendKernelTrack(model, opts.Channel, local); err != nil {
```

Comment:
Should we catch and report conflicts here?  I mean, the channel in opts might indicate a track that is not the one the kernel is coming from. Don't recall how this is used, so feel free to move on with what seems sensible.

Reply:
Its a good question, we can be conservative and simply forbid it. OTOH I can see people doing:
```
snap prepare-image --channel=16/stable core-16.model
```
where everything comes out of the "16" track except for the kernel which is explicitly selected via the `kernel-track`. It seems to me this is an ok use-case. I can add a warning of course.